test/agenda.js

```diff
@@ -1182,6 +1182,33 @@ describe("agenda", function() {
         }, 100);
       });
 
+      it('should run the higher priority jobs first', function (done) {
+        var results = [];
+        var now = new Date();
+
+        jobs.define('priority', { concurrency: 1 }, function (job, cb) {
+          results.push(job.attrs.priority);
+          setTimeout(cb, 50);
+        });
+
+        var normal = jobs.create('priority').schedule(now);
+        var low = jobs.create('priority').schedule(now).priority('low');
+        var high = jobs.create('priority').schedule(now).priority('high');
+
+        normal.save(function () {
+          low.save(function () {
+            high.save(function () {
+              jobs.start();
+
+              setTimeout(function () {
```

Comment:
Thanks @simison and @lushc I'm implementing this right now using `removeListener` as well (for the count = 3 approach). If it doesn't get a count of 3 after X period of time (whatever is default mocha timeout) then it will fail ofc.  See my PR in #462 for the tests to be added there.  Thank you.